shared/macros-oval.jinja

```diff
@@ -515,3 +515,72 @@
 
   </def-group>
 {{%- endmacro %}}
+
+{{#
+  Macro which generates OVAL definition, test and object that check if 
+  a value of certain type is present in a YAML (or JSON) file at a given path.
+  Parameters:
+    - path - path of the file to check
+    - file - name of the file to check
+    - yamlpath - OVAL's YAML Path expression
+    - value - the value to check
+    - type (string/SimpleDatatypeEnumeration) - datatype for state's value_of, default is 'string'
+    - check (string/CheckEnumeration) - entity_check value for state's value_of, default is 'all'
+    - negate (boolean) - invert the meaning of the value check criterion, default false
+    - ocp_data (boolean) - the file is the part of the dump of OCP configuration, take into account the 'ocp_data_root' variable
+#}}
+{{%- macro oval_yamlfile_value(path='', file='', yamlpath='', value='', type='', check='', negate=false, ocp_data=false) -%}}
+{{% if target_oval_version >= [5, 11] %}}
+  <def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="1">
+      <metadata>
+        <title>Check the value of '{{{ yamlpath }}}' in the "{{{ file }}}" file</title>
+        {{{- oval_affected(products) }}}
+        <description>The file {{{ file }}} in {{{ path }}} should {{% if negate %}}not{{% endif %}} contain value "{{{ value }}}" of type {{{ type }}} at {{{ yamlpath }}}.</description>
+      </metadata>
+      <criteria>
+          <criterion comment="Make sure that {{% if negate %}}no{{% endif %}} {{% if check %}}({{{ check }}}){{% endif %}} element of '{{{ yamlpath }}}' has value '{{{ value }}}'' of type {{{ type }}}." {{% if negate %}}negate="true"{{% endif %}} test_ref="test_{{{ rule_id }}}"/>
+          <criterion comment="Make sure that there is the file '{{{ file }}}' exists in {{{ path }}}." test_ref="test_file_for_{{{ rule_id }}}"/>
```

Comment:
What is the reason for this extra test for `{{{ file }}}` existence?
Won't the `yamlfilecontent_test` already fail if it doesn't exist?

Reply:
The test won't find any objects. Sometimes you want the whole rule to fail is objects are found, so the inability to fetch the actual yaml file would cause a false negative.

Reply:
Right, so the check for the file is to be sure that when you don't objects to be found, it is not because the file is missing.

Reply:
Maybe it would make sense to add it only for `negate=true` case.